thunx-pdp-opa/src/main/java/com/contentgrid/thunx/pdp/opa/QuerySetToThunkExpressionConverter.java

```diff
@@ -279,6 +287,89 @@ public ThunkExpression<?> visit(Null nullValue) {
 
         @Override
         public ThunkExpression<?> visit(ArrayTerm arrayTerm) {
+            ScalarCollectingRegoVisitor collectingRegoVisitor = new ScalarCollectingRegoVisitor();
+            arrayTerm.getValue().forEach(scalarTerm -> scalarTerm.accept(collectingRegoVisitor));
+            return Scalar.of(collectingRegoVisitor.getScalars());
+        }
+
+        @Override
+        public ThunkExpression<?> visit(SetTerm setTerm) {
+            ScalarCollectingRegoVisitor collectingRegoVisitor = new ScalarCollectingRegoVisitor();
+            setTerm.getValue().forEach(scalarTerm -> scalarTerm.accept(collectingRegoVisitor));
+            return Scalar.of(collectingRegoVisitor.getScalars());
+        }
+    }
+
+    static class ScalarCollectingRegoVisitor implements RegoVisitor<ThunkExpression<?>> {
+
+        @Getter
+        List<Scalar<?>> scalars = new ArrayList<>();
+
+        @Override
+        public ThunkExpression<?> visit(QuerySet querySet) {
+            return null;
```

Comment:
All these rego types are unexpected in a scalar collection, so they should throw an exception instead of silently dropping the term.

For example, in rego, you can have a `Ref` in an array, for example, when constructing `input.entity.something in [input.entity.optionA, input.entity.optionB]`

That is probably (I didn't check) a valid rego expression, but is not supported as entry in the current `CollectionValue`.

It's fine to limit ourselves to only scalar values for now, but we should not drop those terms silently.

Reply:
I've added support of collection in collection for opa-client (https://github.com/xenit-eu/opa-java-client/pull/28) and for thunx